### DISCUSSION TITLE
ssl证书添加方式

### DISCUSSION THREAD

**Discussion by skyoct** [labring/laf#164] [Generals]
### 上下文
由于openrestry配置不灵活，无法简单动态增加路由等，采用新网关apisix来代替，目前想要给所有的路由增加https，apisix是支持使用api方式添加ssl证书，可以由gateway-controller来调用添加证书到apisix中。
### 问题
gateway-controller如何获取证书？
### 解决方式
1. 手动挂载证书，可能存在需要经常手动更换证书？如何检查证书有效期？
2. 新建一个服务，底层调用acme.sh自动生成证书？
3.

**Comment by skyoct** [comment]
用户对自己添加证书的需求度有多高？

**Reply by maslow** [reply]
平台的证书，还是平台自己来提供， 像原 lafyun 那样。 暂时不用 acme.sh 自动生成证书
